android/libraries/rib-base/src/test/java/com/badoo/ribs/core/routing/configuration/ConfigurationFeatureTest.kt

```diff
@@ -0,0 +1,663 @@
+    }
+
+    @Before
+    fun setUp() {
+        helperPermanent1 =
+            ConfigurationTestHelper.create(Permanent1,2, ViewAttachMode.PARENT)
+
+        helperPermanent2 =
+            ConfigurationTestHelper.create(Permanent2,3, ViewAttachMode.PARENT)
+
+        helperContentViewParented1 =
+            ConfigurationTestHelper.create(ContentViewParented1,2, ViewAttachMode.PARENT)
+
+        helperContentViewParented2 =
+            ConfigurationTestHelper.create(ContentViewParented2,3, ViewAttachMode.PARENT)
+
+        helperContentViewParented3 =
+            ConfigurationTestHelper.create(ContentViewParented3,2, ViewAttachMode.PARENT)
+
+        helperContentExternal1 =
+            ConfigurationTestHelper.create(ContentExternal1,2, ViewAttachMode.EXTERNAL)
+
+        helperContentExternal2 =
+            ConfigurationTestHelper.create(ContentExternal2,3, ViewAttachMode.EXTERNAL)
+
+        val helpers = listOf(
+            helperPermanent1,
+            helperPermanent2,
+            helperContentViewParented1,
+            helperContentViewParented2,
+            helperContentViewParented3,
+            helperContentExternal1,
+            helperContentExternal2
+        )
+
+        resolver = mock {
+            helpers.forEach { helper ->
+                on { invoke(helper.configuration) } doReturn helper.routingAction
+            }
+        }
+
+        poolInTimeCapsule = mapOf(
+            Permanent(0) to Unresolved<Configuration>(SLEEPING, Permanent1, helperPermanent1.bundles),
+            Permanent(1) to Unresolved<Configuration>(SLEEPING, Permanent2, helperPermanent2.bundles),
+
+            Content(0) to Unresolved<Configuration>(SLEEPING, ContentViewParented1, helperContentViewParented1.bundles),
+            Content(1) to Unresolved<Configuration>(SLEEPING, ContentViewParented2, helperContentViewParented2.bundles),
+            Content(2) to Unresolved<Configuration>(INACTIVE, ContentViewParented3, helperContentViewParented3.bundles),
+
+            Content(3) to Unresolved<Configuration>(SLEEPING, ContentExternal1, helperContentExternal1.bundles),
+            Content(4) to Unresolved<Configuration>(INACTIVE, ContentExternal2, helperContentExternal2.bundles)
+        )
+        emptyTimeCapsule = mock()
+        restoredTimeCapsule = mock {
+            on { get<SavedState<Configuration>>(ConfigurationFeature::class.java.name) } doReturn SavedState(
+                pool = poolInTimeCapsule
+            )
+        }
+        parentNode = mock()
+    }
+
+    private val permanentParts = listOf(
+        Permanent1,
+        Permanent2
+    )
+
+    private fun createFeature(timeCapsule: TimeCapsule<SavedState<Configuration>>): ConfigurationFeature<Configuration> {
+        return ConfigurationFeature(
+            initialConfigurations = permanentParts,
+            timeCapsule = timeCapsule,
+            resolver = resolver,
+            parentNode = parentNode
+        )
+    }
+
+    private fun empty() {
```

Comment:
maybe rename to `createEmptyFeature()`/`createRestoredFeature()`?

Reply:
Done